test/model/DeliveryAssemblyServiceTest.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoDeliveryRdf\test\model;
+
+use oat\oatbox\filesystem\FileSystemService;
+use oat\tao\test\TaoPhpUnitTestRunner;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoDeliveryRdf\model\SimpleDeliveryFactory;
+use Prophecy\Argument;
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+class DeliveryAssemblyServiceTest extends TaoPhpUnitTestRunner
+{
+    protected $adapterFixture;
+    protected $sampleDir;
+
+    public function setUp()
+    {
+        \common_ext_ExtensionsManager::singleton()->getExtensionById('taoDeliveryRdf');
+        $this->adapterFixture = 'adapterFixture';
+        $this->sampleDir = __DIR__ . '/../samples/';
+    }
+
+    /**
+     * @dataProvider getResultsOfDeletion()
+     *
+     * @param $runtimeDelete
+     * @param $directoryDelete
+     * @param $resourceDelete
+     * @param $expected
+     */
+    public function testDeleteInstance($runtimeDelete, $directoryDelete, $resourceDelete, $expected)
+    {
+        $smProphecy = $this->prophesize(\core_kernel_classes_Resource::class);
+        $smProphecy->delete()->willReturn($resourceDelete);
+        $delivery = $smProphecy->reveal();
+
+        $deliveryAssemblyServiceMock = $this->getMockBuilder(DeliveryAssemblyService::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('deleteDeliveryRuntime', 'deleteDeliveryDirectory'))
+            ->getMock();
+
+        $deliveryAssemblyServiceMock
+            ->method('deleteDeliveryRuntime')
+            ->with($this->equalTo($delivery))
+            ->will($this->returnValue($runtimeDelete));
+
+        $deliveryAssemblyServiceMock
+            ->method('deleteDeliveryDirectory')
+            ->with($this->equalTo($delivery))
+            ->will($this->returnValue($directoryDelete));
+
+        $this->assertEquals($expected, $deliveryAssemblyServiceMock->deleteInstance($delivery));
```

Comment:
This test checks only returned value, but it does not check whether necessary methods has been called and resources with folders has been deleted from storage. 

For example by the logic of `deleteInstance()` method  if `deleteDeliveryRuntime()` returns `false` then `deleteDeliveryDirectory()` and `$assembly->delete()` should not be called (which is not good idea too, because there should be deleted all or nothing).